src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -78,6 +78,13 @@ @interface MTRDeviceControllerFactoryParams ()
 
 @end
 
+class MTRApplicationCallback : public app::ReadHandler::ApplicationCallback {
+    CHIP_ERROR OnSubscriptionRequested(app::ReadHandler & readHandler, Transport::SecureSession & secureSession) override
+    {
+        return readHandler.SetMaxReportingInterval(kSubscriptionMaxIntervalPublisherLimit);
```

Comment:
This doesn't look right.  If the client asked for a MaxInterval larger that kSubscriptionMaxIntervalPublisherLimit, why would we force a smaller one?